master/buildbot/reporters/base.py

```diff
@@ -76,9 +72,12 @@ def stopService(self):
         for consumer in self._event_consumers:
             yield consumer.stopConsuming()
         self._event_consumers = []
-        if self._pending_got_event_call is not None:
-            yield self._pending_got_event_call
-        self._pending_got_event_call = None
+
+        for pending_got_event_call in list(
+            self._pending_got_event_calls.values()
```

Comment:
I think this could fit on a single line and `pending_got_event_call` could be named pending_call to shorten the line as well.